client/me/purchases/vat-info/index.tsx

```diff
@@ -5,6 +5,7 @@ import React, { useState, useEffect } from 'react';
 import { useTranslate } from 'i18n-calypso';
 import { useDispatch } from 'react-redux';
 import { CompactCard, Button, Card } from '@automattic/components';
+import { recordTracksEvent } from '@automattic/calypso-analytics';
```

Comment:
Here you're importing the raw `recordTracksEvent` but that version does not require a redux dispatch and [the README says not to use it if you can](https://github.com/Automattic/wp-calypso/blob/37b6bf1332e3ee7ba149f99df4d7506afa82eda5/packages/calypso-analytics/README.md#L19) I think you actually want:

```
import { recordTracksEvent } from 'calypso/state/analytics/actions';
```